librz/analysis/var.c

```diff
@@ -151,21 +151,51 @@ RZ_API RZ_BORROW RzAnalysisVar *rz_analysis_function_set_var(RzAnalysisFunction
 }
 
 /**
- * Add or update a variable \p var to the given function \p fcn.
+ * Add or update a variable \p var to the given function \p fcn. Will not update existing DWARF variables
  *
  * \param fcn the function which the variable will belong to
  * \param var the variable to add or update
  * \param size \p var's type size
  * \return the created or updated variable, or NULL if the operation could not be completed
  */
-RZ_API RZ_BORROW RzAnalysisVar *rz_analysis_function_add_var(RzAnalysisFunction *fcn, RZ_OWN RzAnalysisVar *var,
+RZ_IPI RZ_BORROW RzAnalysisVar *rz_analysis_function_add_var_dwarf(RzAnalysisFunction *fcn, RZ_OWN RzAnalysisVar *var,
```

Comment:
I don't understand what it means that this function "will not update existing DWARF variables". Also, why the `_dwarf` in the function name?

Reply:
For the variable information in DWARF, there may be some variables whose location information is the same
but other information may be completely different..

The current approach to this situation is to add all `FORMAL_PARAMETER` variables first, and then add the others. This
assumes that for all `FORMAL_PARAMETER` variables of a function, there is no overlap in their location.

So this function is written specifically for this case, so it has a `_dwarf` suffix.